### DISCUSSION TITLE
Swagger does not display the types of Query/Cookie parameters when they are Optional

### DISCUSSION THREAD

**Discussion by BrentHuang** [fastapi/fastapi#11901] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, Optional
import uuid
from fastapi import Cookie, FastAPI, Path, Query, Response

app = FastAPI()


@app.get("/items/{item_id}")
async def read_items(
    item_id: Annotated[int, Path(title="The ID of the item to get")],
    x: Annotated[Optional[str], Query()],  # 套在 Annotated 中时，在 Query 中通过 default=None 指定 x 的默认值是无效的，FastAPI 会报错
    q: Annotated[Optional[str], Query(alias="item-query")] = None,  # q 有默认值 None，是可选的
    a: Optional[int] = Query(default=None)):  # 在 = 右边的 Query 中用 default 指定默认值，a 就是可选的
    return {'item_id': item_id, 'x': x, 'q': q, 'a': a}


@app.get("/setcookie")
def set_cookie(response: Response):  # todo 路径操作函数有一个 response 参数？
    # 与 cookie 相关的两个函数 delete_cookie 和 set_cookie。set_cookie 还有其它的 cookie 相关参数，如 max_age、expires
    response.delete_cookie(
        'cookie1'
    )  # 效果是在 response headers 中增加一个 header: Set-Cookie: cookie1=""; expires=Sat, 27 Jul 2024 06:32:08 GMT; Max-Age=0; Path=/; SameSite=lax
    response.set_cookie(
        "ads_id",
        uuid.uuid4().hex
    )  # 效果是在 response headers 中增加一个 header: Set-Cookie: ads_id=0b74a4f9a20f4ea5aaa826b1d6fbc9d0; Path=/; SameSite=lax
    response.set_cookie('x2', 123)
    return {"result": "set cookie ok"}


# 后续的请求中会带上 Cookie: ads_id=704ad7ec7c0640368cf26337abeedd83; x2=123 这个 http header
@app.get("/items")
async def read_items(
    ads_id: Annotated[Optional[str], Cookie()] = None,
    x2: Annotated[Optional[int],
                  Cookie()] = None):  # 注意这里的参数，如果将 Optional 写在 Annotated 的外面，则 OpenAPI 无法识别为 Cookie 参数，而是当做 Query 参数
    return {"ads_id": ads_id, 'x2': x2}
```

### Description

Swagger does not display the types of Query/Cookie parameters when they are Optional:
![image](https://github.com/user-attachments/assets/b355dc8f-6627-4062-a012-6604221893e5)
![image](https://github.com/user-attachments/assets/235c8431-67a9-4fd0-bf3a-1b3550cd8ab9)

Expect something like this:
![image](https://github.com/user-attachments/assets/f9b866e6-6d3d-432b-a4b7-dcf4d030bcb5)

### Operating System

Windows

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

Python 3.12.1

**Comment by YuriiMotov** [comment, marked as answer]
Are you sure you use FastAPI `0.111.1`?

This problem is related to previous versions and was solved with `0.111.1`. Now FastAPI uses the latest version of `swagger-ui`.
I just checked your code, swagger shows correct types of parameters.

Try opening the html-code of swagger page and look at the version of `swagger-ui`:

```
<script src="https://cdn.jsdelivr.net/npm/swagger-ui-dist@5/swagger-ui-bundle.js"></script>
```

If your version is different, check:

1) You use the latest version of FastAPI

2) You DO NOT use self-hosted `swagger-ui`([docs](https://fastapi.tiangolo.com/how-to/custom-docs-ui-assets/)). If you DO use it, update local files to latest versions

3) Try clearing web-browser cache and reload swagger page

**Comment by BrentHuang** [comment]
thank you. 

I use self-hosted swagger-ui, but not the latest versions.
